server/auth/auth_oauth.py

```diff
@@ -226,14 +226,13 @@ def get_tokens(self):
             else:
                 value = request.cookies.get(self.cookie_params["key"])
                 value = base64.b64decode(value)
-                try:
-                    tokensdict = json.loads(value)
-                    g.tokens = Tokens(**tokensdict)
-                except (TypeError, KeyError, json.decoder.JSONDecodeError):
-                    g.pop("tokens", None)
-                    return None
+                tokensdict = json.loads(value)
+                g.tokens = Tokens(**tokensdict)
 
-        except (TypeError, KeyError):
+        except Exception:
+            # there are many types of exceptions that can be raise in the above section.
+            # It is impractical to list all the exceptions here, since that would be brittle.
+            # If an exception occurs, then return None, meaning that no token could be retrieved.
```

Comment:
Should we be logging the error to expose if our users are somehow being served invalid cookies?

Reply:
Good idea, I'll add that.